### PR TITLE
Update CONTRIBUTION-FORMULA.md

### PR DESCRIPTION
**Pull request by craigbox**
Per Steering discussion

### CODE REVIEW

File: steering/CONTRIBUTION-FORMULA.md [hunk 1]
```diff
@@ -4,13 +4,12 @@ The [steering committee charter](CHARTER.md) states "Each year, when
...
 allocation of seats.
 
-Attribution is done using the company attribution table used by the
-istio-policy-bot. Requests for changes to attribution can be made by e-mail
-to elections@discuss.istio.io.
+Attribution is done using the [cncf/gitdm](https://github.com/cncf/gitdm) developer affiliation files.
+Changes to attribution can be made by pull request to that repository.
 
-[The method and results are published here](https://docs.google.com/spreadsheets/d/1Dt-h9s8G7Wyt4r16ZVqcmdWXDuCaPC0kPS21BuAfCL8/edit#gid=0).
+[The method and results for 2021-2022 are published here](https://docs.google.com/spreadsheets/d/187LDk96ggccHa_IBBGLEHYEBmR5eidsEQUDy2nE9kxo/edit).
```
Comment (howardjohn) [inline steering/CONTRIBUTION-FORMULA.md:15]: The tabs still show "PRs merged" as the criteria on the spread sheet

File: steering/CONTRIBUTION-FORMULA.md [hunk 2]
```diff
@@ -4,13 +4,12 @@ The [steering committee charter](CHARTER.md) states "Each year, when
 Contribution Seat terms begin, the Steering Committee will vote on an exact
 formula and procedure for determining allocation for the following year."
 
-For the February 2021 to January 2022 term, the contribution formula shall be
-**merged pull requests** attributed to a Contributing Company,
+For the February 2022 to January 2023 term, the contribution formula shall be
+**contributions** (review, comment, commit, issue or PR on GitHub) attributed to a Contributing Company,
```
Comment (ZackButcher) [inline steering/CONTRIBUTION-FORMULA.md:8]: Can you point at the script we're using to calculate (since it diverges from the github contribution stat)?

**Review by ZackButcher — COMMENTED** [review]
Looks good to me overall, and a more broad measure of contribution than purely PRs is a good thing for the community overall I think.

File: steering/CONTRIBUTION-FORMULA.md (hunk 2 above)
Comment (howardjohn) [reply steering/CONTRIBUTION-FORMULA.md:8]: Also interested in a diff of the results of old vs new methodology to see if there are any meaningful differences

File: steering/CONTRIBUTION-FORMULA.md (hunk 1 above)
Comment (ZackButcher) [reply steering/CONTRIBUTION-FORMULA.md:15]: That's correct - we used PRs merged for 21-22 (this proposal is for 22-23 and onwards).

File: steering/CONTRIBUTION-FORMULA.md (hunk 2 above)
Comment (hzxuzhonghu) [reply steering/CONTRIBUTION-FORMULA.md:8]: Multi dimensions could help motivate the community activities. I think it is better than before.

Comment (craigbox) [reply steering/CONTRIBUTION-FORMULA.md:8]: > Also interested in a diff of the results of old vs new methodology to see if there are any meaningful differences

Screen Shot 2022-04-01 at 2 22 42 PM

This isn't on the exact same time period, but I think it produces an acceptably similar outcome to the current formula, while accurately reflecting contribution in other areas of the project.

Comment (hzxuzhonghu) [reply steering/CONTRIBUTION-FORMULA.md]: This is a hyperthesis, huawei will keep the contributions until 1.31, 2023.

Comment (howardjohn) [reply steering/CONTRIBUTION-FORMULA.md]: I would note that 80% of "contributions" are from comments. A comment has much less value than something like a PR or an issue (for example, I just made a contribution by writing this). Especially with things like `/retest`, etc.

The end result is almost the same as PRs, so it's not a big concern in the current allocations - just something that could easily be gamed if someone wanted to.

No metric is perfect, so I'm not suggesting we change it

Comment (craigbox) [reply steering/CONTRIBUTION-FORMULA.md]: This was discussed.  Sustained low-effort comments in order to try and game the system would be addressed through the Code of Conduct, and could result in people being removed from the community.

I like the fact that right now contributions effectively have to be approved by another community member (as we're counting merged PRs that passed review), but we are trying to incentivize PR review also.

We could remove comments and see the outcome?  Let's give that a go...

Comment (hzxuzhonghu) [reply steering/CONTRIBUTION-FORMULA.md]: Assume pr review comments and the comments here are not the same. But think of comments under an issue., which may be of great help for end users

Comment (craigbox) [reply steering/CONTRIBUTION-FORMULA.md]: Screen Shot 2022-04-01 at 3 01 47 PM

Same result, slightly different way of getting there.

Comment (craigbox) [reply steering/CONTRIBUTION-FORMULA.md]: Turns out what we're seeing measured is "PR review comments", not "PR reviews".  Devstats doesn't measure reviews.  See https://github.com/cncf/devstats/issues/341

Comment (hzxuzhonghu) [reply steering/CONTRIBUTION-FORMULA.md]: Personally i think "PR review comments" may be more fair, as one pr can be super huge and one could be small, the review comments number should be different based on the effort

**Comment by craigbox** [comment]
Calling a Steering vote; please do so by +1 on this issue.

**Review by linsun — APPROVED** [review]
LGTM

**Review by hzxuzhonghu — APPROVED** [review]

**Review by irisdingbj — APPROVED** [review]

**Review by kebe7jun — APPROVED** [review]

**Comment by rcernich** [comment]
LGTM, +1

**Review by rcernich — APPROVED** [review]

**Review by ZackButcher — APPROVED** [review]

**Comment by angaluri** [comment]
LGTM

**Comment by craigbox** [comment]
Vote passes.

**Comment by angaluri** [comment]
Have we explored a weighted formula for merged, reviewed, and commented for
example? Then we can weight comments lower than others. I also like the
idea of eliminating comments from being counted.

Srihari

**Pull request merged**